bpf/lib/subnet.h

```diff
@@ -0,0 +1,94 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/* Copyright Authors of Cilium */
+
+#pragma once
+
+#include "common.h"
+
+#include <linux/ip.h>
+#include "ipv6.h"
+
+#define SUBNET_MAP_SIZE 1024
+
+struct subnet_key {
+	struct bpf_lpm_trie_key lpm_key;
+	__u16 pad2;
+	__u8 pad1;
+	__u8 family;
+	union {
+		struct {
+			__u32		ip4;
+			__u32		pad4;
+			__u32		pad5;
+			__u32		pad6;
+		};
+		union v6addr	ip6;
+	};
+} __packed;
+
+struct subnet_value {
+    __u32 identity;
+};
+
+/* CIDR -> Identity map */
```

Comment:
Nit: The term `identity` is very confusing (there's an another concept for the NetworkPolicy). I'd prefer more explicit name like `subnet_id`.